File: apps/api-server/src/main/java/com/audioguide/entity/Shop.java
package com.audioguide.entity;


import com.audioguide.enums.Status;
import jakarta.persistence.*;
import lombok.*;
import lombok.experimental.FieldDefaults;

import java.time.LocalDate;

@Entity
@Table(name = "shop")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@FieldDefaults(level = AccessLevel.PRIVATE)
public class Shop {


    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    Integer id;


    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "owner_id", nullable = false)
    User owner;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "shop_type_id", nullable = false)
    ShopType shopType;


    String name;
    String  address;
    @Column(name = "short_description")
    String shortDescription;
    String description;
    String imageName;
    String audioURL;
    Double lat;
    Double lng;

    Integer avgCostPerPerson;
    Integer avgWaitTimeMin;   // in minutes
    Integer avgEatTimeMin;
    LocalDate createdAt;
    @Enumerated(EnumType.STRING)
    Status status;

}

File: apps/customer-web/src/types/shop.ts
export type ShopResponse = {
  id: number;
  ownerId: string;
  name: string;
  address: string;
  shortDescription?: string;
  detailedDescription?: string;
  description: string;
  imageName: string;
  audioURL: string;
  shopTypeId?: number;
  shopTypeName?: string;
  lat: number;
  lng: number;
  avgCostPerPerson: number;
  avgWaitTimeMin: number;
  avgEatTimeMin: number;
  createdAt: string;
  status: string;
};

export type ShopCreationRequest = {
  name: string;
  address: string;
  shortDescription?: string;
  detailedDescription?: string;
  description: string;
  lat: number;
  lng: number;
  avgCostPerPerson: number;
  avgWaitTimeMin: number;
  avgEatTimeMin: number;
};

export type ShopNarrationResponse = {
  shopId: number;
  language: string;
  languageKey?: string;
  requestedLanguage?: string;
  voice: string;
  sourceText?: string;
  script?: string;
  audioUrl: string;
  cached: boolean;
  fallbackApplied?: boolean;
  updatedAt?: string;
};
